esp/services/ws_dfu/ws_dfuService.cpp

```diff
@@ -1412,20 +1420,16 @@ bool CWsDfuEx::DFUDeleteFiles(IEspContext &context, IEspDFUArrayActionRequest &r
     }
 
     StringBuffer returnStr;
-    StringBuffer auditStr(",FileAccess,WsDfu,DELETED,");
     IArrayOf<IEspDFUActionInfo> actionResults;
 
     StringArray superFiles, failedFiles;
 
-    auditStr.append(espProcess.get());
-    auditStr.append(',');
-    if (!isEmptyString(username))
-        auditStr.append(username).append('@');
-    context.getPeer(auditStr);
-
-    doDeleteSuperFiles(req.getLogicalFiles(), userdesc, superFiles, failedFiles, auditStr.str(),
+    VStringBuffer auditStr(",FileAccess,Esp,%s,DELETE", espProcess.get());
```

Comment:
It feels like we should either enumerate the actions (READ|CREATE|DELETE|COPY), or at least #Define them to mitigate possible errors in the future

Reply:
honestly the whole thing should be reworked.. components should not be building these strings up like this manually..
You could do 10% of job now though, I have no objection.

Reply:
Perhaps worth biting the bullet and adding a function and some enums etc. to build up a well formatted/consistent audit log.. 
but that is more work of course and more changes to everywhere else.

Reply:
I think I'm 100% in agreement that this should be handled in a centralized function... how are the other components constructing the audit entry?